src/entries/popup/pages/home/TokenDetails/TokenDetails.tsx

```diff
@@ -267,9 +273,58 @@ export const getCoingeckoUrl = ({
   return `https://www.coingecko.com/en/coins/${mainnetAddress || address}`;
 };
 
-function MoreOptions({ token }: { token: ParsedUserAsset }) {
+function MoreOptions({
+  token,
+  swappable,
+}: {
+  token: ParsedUserAsset;
+  swappable: boolean;
+}) {
+  const { hiddenAssets, removeHiddenAsset, addHiddenAsset } =
+    useHiddenAssetStore();
+
+  const { pinnedAssets, removedPinnedAsset, addPinnedAsset } =
+    usePinnedAssetStore();
+
+  const isHidden = useCallback(
+    (asset: ParsedUserAsset) =>
+      hiddenAssets.some(
+        (uniqueId) => uniqueId === computeUniqueIdForHiddenAsset(asset),
+      ),
+    [hiddenAssets],
+  );
+
+  const hidden = isHidden(token);
   const explorer = getTokenBlockExplorer(token);
   const isNative = isNativeAsset(token.address, token.chainId);
+  const pinned = pinnedAssets.some(
+    ({ uniqueId }) => uniqueId === token.uniqueId,
+  );
+
+  const toggleHideToken = useCallback(() => {
+    if (hidden) {
+      removeHiddenAsset({ uniqueId: computeUniqueIdForHiddenAsset(token) });
+      return;
+    }
+    if (pinned) removedPinnedAsset({ uniqueId: token.uniqueId });
+    addHiddenAsset({ uniqueId: computeUniqueIdForHiddenAsset(token) });
+  }, [
+    token,
+    hidden,
+    pinned,
+    removedPinnedAsset,
+    addHiddenAsset,
+    removeHiddenAsset,
+  ]);
+
+  const togglePinToken = useCallback(() => {
```

Comment:
if your not passing this function to a memoized component it don't need keep it's ref stable (`useCallback`)

Reply:
because like it is recreating the function anyway, just going thru more effort to check the cache and return the previous if deps didn't change
I mean `useCallback` is super cheap but not free

not a change request, just a note